core/consensus/babe/impl/babe_config_repository_impl.cpp

```diff
@@ -111,24 +115,34 @@ namespace kagome::consensus::babe {
       warp(lock, finalized);
     }
 
-    auto genesis_res = config({block_tree_->getGenesisBlockHash(), 0}, false);
-    if (not genesis_res) {
-      SL_ERROR(logger_, "get config at genesis error: {}", genesis_res.error());
-      return false;
+    if (!timings_) {
+      auto genesis_res = config({block_tree_->getGenesisBlockHash(), 0}, false);
+      if (genesis_res.has_value()) {
+        auto &genesis = genesis_res.value();
+        timings_.init(genesis->slot_duration, genesis->epoch_length);
+        SL_DEBUG(logger_,
+                 "Timing was initialized: slot is {}ms, epoch is {} slots",
+                 timings_.slot_duration.count(),
+                 timings_.epoch_length);
+      }
     }
-    auto &genesis = genesis_res.value();
-    slot_duration_ = genesis->slot_duration;
-    epoch_length_ = genesis->epoch_length;
 
+    [[maybe_unused]] bool active_ = false;
+
+    // TODO Perhaps, should be observed all non-finalized blocks
     auto best = block_tree_->bestBlock();
-    auto best_header = block_tree_->getBlockHeader(best.hash).value();
-    if (auto res = config(best, true); not res and not config_warp_sync_) {
-      SL_ERROR(logger_, "get config at best {} error: {}", best, res.error());
-      if (not trie_storage_->getEphemeralBatchAt(best_header.state_root)) {
-        SL_ERROR(logger_,
-                 "warp sync was not completed, restart with \"--sync Warp\"");
+    auto consensus = consensus_selector_.get()->getProductionConsensus(best);
+    if (std::dynamic_pointer_cast<Babe>(consensus)) {
+      active_ = true;
+      if (auto res = config(best, true); not res and not config_warp_sync_) {
+        SL_ERROR(logger_, "get config at best {} error: {}", best, res.error());
+        auto best_header = block_tree_->getBlockHeader(best.hash).value();
+        if (not trie_storage_->getEphemeralBatchAt(best_header.state_root)) {
+          SL_ERROR(logger_,
+                   "warp sync was not completed, restart with \"--sync Warp\"");
+        }
+        return false;
       }
-      return false;
     }
```

Comment:
Should we proceed if `consensus` is some unknown implementation or it is initialized by nullptr (which could be the case for `getProductionConsensus` returned value)? 
Probably makes sense adding else branch that handles such cases